R/shapley_setup.R

```diff
@@ -139,20 +171,31 @@ shapley_setup <- function(internal) {
 #'
 #' # Subsample of coalitions
 #' x <- create_coalition_table(exact = FALSE, m = 10, n_coalitions = 1e2)
-create_coalition_table <- function(m, exact = TRUE, n_coalitions = 200, weight_zero_m = 10^6,
-                                   paired_shap_sampling = TRUE, prev_coal_samples = NULL,
+create_coalition_table <- function(m,
+                                   exact = TRUE,
+                                   n_coalitions = 200,
+                                   weight_zero_m = 10^6,
+                                   paired_shap_sampling = TRUE,
+                                   prev_coal_samples = NULL,
                                    coal_feature_list = as.list(seq_len(m)),
                                    approach0 = "gaussian",
-                                   shapley_reweighting = "none") {
+                                   shapley_reweighting = "none",
+                                   dt_valid_causal_coalitions = NULL) {
   if (exact) {
-    dt <- exact_coalition_table(m, weight_zero_m)
+    dt <- exact_coalition_table(
+      m = m,
+      weight_zero_m = weight_zero_m,
+      dt_valid_causal_coalitions = dt_valid_causal_coalitions
```

Comment:
not superimportant, but we could maybe call this dt_valid_coalitions, just add a comment on this line and in the documentation that it is only relevant for for causal?

Reply:
same for the sampling version

Reply:
I discussed it with Martin, and we decided to let it be like this for now.